packages/wrangler/src/config/validation.ts

```diff
@@ -182,11 +182,18 @@ export function normalizeAndValidateConfig(
 
 	if (envName !== undefined) {
 		if (isRedirectedConfig) {
-			diagnostics.errors.push(dedent`
-				You have specified the environment "${envName}", but are using a redirected configuration, produced by a build tool such as Vite.
-				You need to set the environment in your build tool, rather than via Wrangler.
-				For example, if you are using Vite, refer to these docs: https://developers.cloudflare.com/workers/vite-plugin/reference/cloudflare-environments/
-			`);
+			const isPagesExpected =
+				isPagesConfig(rawConfig) && ["production", "preview"].includes(envName);
```

Comment:
What's the second check here for? Pages projects should only ever have that lmited set of environments

Reply:
mh... yeah no, a totally fair point, I'll remove it 👍

Reply:
removed 🙂